trakt/sync.py

```diff
@@ -288,19 +288,26 @@ def search_by_id(query, id_type='imdb', media_type=None, slugify_query=False):
         if 'episode' in d:
             from trakt.tv import TVEpisode
             show = d.pop('show')
-            extract_ids(d['episode'])
+            episode = d.pop('episode')
+            extract_ids(episode)
             results.append(TVEpisode(show.get('title', None),
                                      show_id=show['ids'].get('trakt'),
```

Comment:
For consistancy, you should extract show ids just like in users.py

Reply:
what do you mean? using `item['type']`?

Reply:
extract_ids(show)

show_id=show.get('trakt')

Reply:
This then: ea690b64ecfed63354a225840d097a9a591635ca